test_scripts/Resumption/Handling_errors_from_HMI/050_resume_IVD_subscription_after_erroneous_response_all_moduleTypes.lua

```diff
@@ -0,0 +1,104 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0190-resumption-data-error-handling.md
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0188-get-interior-data-resumption.md
+--
+-- Description:
+-- In case:
+-- 1. Subscriptions for all interior vehicle data types are added
+-- 2. Unexpected disconnect and Reconnect are performed
+-- 3. App reregisters with actual HashId
+-- 4. HMI responds with erroneous resultCode for one interior vehicle data type
+-- SDL does:
+-- 1. process unsuccess response from HMI
+-- 2. remove already restored data
+-- 3. respond RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+-- [[ Local Function ]]
+local function getValue(pModuleType, pSubscribe)
+  local moduleIdValue = common.getModuleControlData(pModuleType, 1).moduleId
+  return { moduleType = pModuleType, moduleId = moduleIdValue, subscribe = pSubscribe }
+end
+
+local function getExpectedData()
+  local expectedData = {}
+  for _, moduleTypeValue in pairs(common.rcModuleTypes) do
+    table.insert(expectedData, getValue(moduleTypeValue, true))
+    table.insert(expectedData, getValue(moduleTypeValue, false))
+  end
+  return expectedData
+end
+
+local function checkResumptionData()
+  local expectedData = getExpectedData()
+  local expectedNumber = 2*#common.rcModuleTypes-1 --`-1` because SDL does not send revert request for data
+                                                   -- with erroneous response
+  common.getHMIConnection():ExpectRequest("RC.GetInteriorVehicleData")
+  :Do(function(exp, data)
+      common.log(data.method .. ", moduleType: " .. data.params.moduleType)
+      if exp.occurences == #common.rcModuleTypes then
+        for k, value in pairs(expectedData) do
+          if common.isTableEqual(value, getValue(data.params.moduleType, false)) then
+            table.remove(expectedData, k)
+          end
```

Comment:
@GetmanetsIrina 
1. Is it really needed to compare all data from `value` and build similar big structure from `data.params.moduleType`?
2 Deleting items from array during iteration by it in forward way can lead to logical errors. 
I would propose next changes to resolve mentioned issues
```suggestion
        for k = #expectedData, 1, -1 do
          if expectedData[k].moduleType == data.params.moduleType and expectedData[k].subscribe == false then
            table.remove(expectedData, k)
          end
```

Reply:
Is it really needed to compare all data from value and build similar big structure from data.params.moduleType?

it's needed to compare moduleType and subscribe parameter and these are 2 params from 3